aws/resource_aws_media_package_channel.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/mediapackage"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsMediaPackageChannel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsMediaPackageChannelCreate,
+		Read:   resourceAwsMediaPackageChannelRead,
+		Update: resourceAwsMediaPackageChannelUpdate,
+		Delete: resourceAwsMediaPackageChannelDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"channel_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[\w-]+$`), "must only contain alphanumeric characters, dashes or underscores"),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Managed by Terraform",
+			},
+			"ingest_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"password": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsMediaPackageChannelCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediapackageconn
+
+	input := &mediapackage.CreateChannelInput{
+		Id:          aws.String(d.Get("channel_id").(string)),
+		Description: aws.String(d.Get("description").(string)),
+	}
+
+	_, err := conn.CreateChannel(input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("channel_id").(string))
+	return resourceAwsMediaPackageChannelRead(d, meta)
+}
+
+func resourceAwsMediaPackageChannelRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mediapackageconn
+
+	input := &mediapackage.DescribeChannelInput{
+		Id: aws.String(d.Id()),
+	}
+	resp, err := conn.DescribeChannel(input)
+	if err != nil {
+		return err
+	}
+	d.Set("arn", resp.Arn)
```

Comment:
The `arn` attribute is not defined in the resource schema and error checking of `d.Set()` is not being performed, so this should currently be silently failing to do anything.

```go
			"arn": {
				Type:     schema.TypeString,
				Computed: true,
			},
```

And can be tested in the basic test via something like:

```go
testAccMatchResourceAttrRegionalARN(resourceName, "arn", "mediapackage", regexp.MustCompile(`channels/.+`)),
```